specification/sections/05-events.md

```diff
@@ -134,6 +136,9 @@ See [provider initialization](./02-providers.md#24-initialization) and [setting
 
 #### Requirement 5.3.3
 
-> `PROVIDER_READY` handlers attached after the provider is already in a ready state **MUST** run immediately.
+> `PROVIDER_READY` handlers attached after the provider is in a ready state **MUST** run immediately.
```

Comment:
@thomaspoignant I think you recently faced some challenges here...

As @Kavindu-Dodan pointed out elsewhere, this point should _probably_ be extended to apply to `STALE` and `ERROR` as well - basically, we don't want race conditions between the provider state and the addition of handlers. I was thinking only of the "happy" path when I originally added this. :sweat_smile:

Reply:
If

Reply:
I support this change. Handler registration can occur in a later stage of an application (ex:- setup phase for provider init and handlers per client in a request scope), so it makes sense to propagate states to new handlers.